framework/core/src/Extend/Conditional.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Extend;
+
+use Flarum\Extension\Extension;
+use Flarum\Extension\ExtensionManager;
+use Illuminate\Contracts\Container\Container;
+
+class Conditional implements ExtenderInterface
+{
+    /**
+     * @var array<array{condition: bool|callable, extenders: ExtenderInterface[]}>
+     */
+    protected $conditions = [];
+
+    /**
+     * @param ExtenderInterface[] $extenders
+     */
+    public function whenExtensionEnabled(string $extensionId, array $extenders): self
```

Comment:
The signature doesn't match the PR comment example. Your example uses a callable as the second argument, not an array. The array is the return response from the callable yes. I assume your example is wrong then..

Also the phpdoc is incorrect, there are two args.

Reply:
yea updated the example thanks

phpdoc doesn't matter, we only add typings for what isn't typed, we can't vanilla type array as ExtenderInterface[], we've being doing this since 1.0 to avoid unnecessary duplication. And the phpdoc standard allows it.